frameworks/AutoGluon/exec.py

```diff
@@ -64,18 +67,30 @@ def run(dataset, config):
             **training_params
         )
 
-    test_data = TabularDataset(test_path)
     # Persist model in memory that is going to be predicting to get correct inference latency
     predictor.persist_models('best')
 
+    def inference_time_classification(data: Union[str, pd.DataFrame]):
+        return None, predictor.predict_proba(data, as_multiclass=True)
+
+    def inference_time_regression(data: Union[str, pd.DataFrame]):
+        return predictor.predict(data, as_pandas=False), None
+
+    infer = inference_time_classification if is_classification else inference_time_regression
+    inference_times = {}
+    if config.measure_inference_time:
+        inference_times["file"] = measure_inference_times(infer, dataset.inference_subsample_files)
+        test_data = pd.read_parquet(dataset.test.path)
+        inference_times["df"] = measure_inference_times(
+            infer,
+            [(1, test_data.sample(1, random_state=i)) for i in range(100)],
+        )
+
+    test_data = TabularDataset(test_path)
+    with Timer() as predict:
+        predictions, probabilities = infer(test_data)
     if is_classification:
-        with Timer() as predict:
-            probabilities = predictor.predict_proba(test_data, as_multiclass=True)
         predictions = probabilities.idxmax(axis=1).to_numpy()
```

Comment:
Note: in v0.8, we now have a cleaner way to do this: https://auto.gluon.ai/dev/api/autogluon.tabular.TabularPredictor.get_pred_from_proba.html

We can probably keep as is for now. Switching will only matter if we end up running on metrics where decision threshold calibration matters (balanced accuracy, f1, etc.), as the current code in AMLB is forcibly using decision_threshold=0.5.

Reply:
We won't use those metrics for our next set of experiments, but PR to make things _also_ work for `0.8` is welcome.